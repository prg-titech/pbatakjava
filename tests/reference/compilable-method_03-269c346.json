{
    "basename": "compilable-method_03-269c346",
    "cmd": "{infile}",
    "infile": "tests/method_03/build.sh",
    "infile_hash": "007d191c26263f17113537fa389af36ecd8ce948604c284f3b3b9e67",
    "stdout": "compilable-method_03-269c346.stdout",
    "stdout_hash": "7dc6465d0cb4b2d6e49cbf7a04978b71b70eedeaaaab1cb9442c1f9a",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}