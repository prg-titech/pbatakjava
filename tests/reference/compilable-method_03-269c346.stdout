
public interface Point {
  public int get__x();

  public int set__x(int toAssign);

  public int get__y();

  public int set__y(int toAssign);

  public Point inverse(Point_Factory V);
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }

  public Point inverse(Point_Factory V) {
    return null;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public Point inverse(Point_Factory V) {
    return V.make(y, x);
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}

public interface Line {
  public Point get__p1();

  public Point set__p1(Point toAssign);

  public Point get__p2();

  public Point set__p2(Point toAssign);

  public Line rotate(Point_Factory Y);
}

public interface Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2);
}

public class Line_v1 implements Line {
  public Point_Factory V;

  public Point_Factory W;

  public Point p1;

  public Point p2;

  public Line_v1(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    this.V = V;
    this.W = W;
    this.p1 = p1;
    this.p2 = p2;
  }

  public Point get__p1() {
    return this.p1;
  }

  public Point set__p1(Point toAssign) {
    return this.p1 = toAssign;
  }

  public Point get__p2() {
    return this.p2;
  }

  public Point set__p2(Point toAssign) {
    return this.p2 = toAssign;
  }

  public Line rotate(Point_Factory Y) {
    return null;
  }
}

public class Line_v1_Factory implements Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    return new Line_v1(V, W, p1, p2);
  }
}

public class Line_v2 implements Line {
  public Point_Factory V;

  public Point_Factory W;

  public Point p1;

  public Point p2;

  public Line_v2(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    this.V = V;
    this.W = W;
    this.p1 = p1;
    this.p2 = p2;
  }

  public Line rotate(Point_Factory Y) {
    return new Line_v1(V, Y, p1, p2.inverse(Y));
  }

  public Point get__p1() {
    return this.p1;
  }

  public Point set__p1(Point toAssign) {
    return this.p1 = toAssign;
  }

  public Point get__p2() {
    return this.p2;
  }

  public Point set__p2(Point toAssign) {
    return this.p2 = toAssign;
  }
}

public class Line_v2_Factory implements Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    return new Line_v2(V, W, p1, p2);
  }
}

public interface Test {
}

public interface Test_Factory {
}

public class Test_v1 implements Test {
  public static void main(String[] args) {
    Line l1 = new Line_v2(new Point_v2_Factory(), new Point_v2_Factory(), new Point_v2(2, 2), new Point_v2(1, 1));
    Line l2 = l1.rotate(new Point_v1_Factory());
  }
}

public class Test_v1_Factory implements Test_Factory {
}
